### DISCUSSION TITLE
Run a simulation from command shell Window or Linux

### DISCUSSION THREAD

**Discussion by jonathanlsn** [Open-Systems-Pharmacology/Forum#918] [Questions & Problems] [answered]
Hello all, I have to run many simulations on Mobi/PK-Sim and I would like to automate the simulation process. I would like to know if there is a way to execute a simulation from Windows or Linux command shell ?

Thank you in advance.
Jonathan

**Comment by Yuri05** [comment, marked as answer]
You can do it via the **OSP R-toolbox** (both under Windows and Linux); s. the [documentaion](https://docs.open-systems-pharmacology.org/working-with-r/r-introduction) for details. 
* R-Toolbox is not part of the OSP Suite setup and must be installed separately. Installation steps depend on the operating system and are described in the documentation. 
* Current version of the R-Toolbox works only with R 3, not with R 4

R-Toolbox works with simulations in **pkml** format. If you have only PK-Sim/MoBi projects - you must export all simulations of interest as pkml.
* For **PK-Sim** it can be done via the _PK-Sim CLI_ (command line interface) - s. the [documentaion](https://docs.open-systems-pharmacology.org/working-with-pk-sim/pk-sim-documentation/pk-sim-command-line-interface) for details. E.g.
  `PKSim.CLI export -p "C:\Temp\ExampleProject.pksim5" -o "C:\Temp\Output" --pkml`
* For **MoBi** you can export simulations as pkml only manually.

**Reply by Yuri05** [reply]
P.S. If you just want to run simulations AS IS and export simulations results, you can do it with PK-Sim CLI only, e.g.
`PKSim.CLI export -p "C:\Temp\ExampleProject.pksim5" -o "C:\Temp\Output" --csv --json -- excel`
As soon as you need any kind of simulation _modifications_ (change parameter values, select other outputs of interest, change output resolution, etc.) or if you need to get more information out of the simulation (e.g. need some knowledge about input parameters etc.) - you would need the R-Toolbox.

**Reply by jonathanlsn** [reply]
Perfect, thank you a lot !
